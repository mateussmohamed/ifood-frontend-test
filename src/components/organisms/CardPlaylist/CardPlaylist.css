.card-playlist {
  width: 100%;
  display: flex;
  flex-direction: row;
  padding: 30px;
}
