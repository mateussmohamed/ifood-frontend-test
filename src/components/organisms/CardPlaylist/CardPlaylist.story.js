import { storiesOf } from '@storybook/react';
import React, { Component } from 'react';
import CardPlaylist from './index';
import Card from '../../molecules/Card';

const stories = storiesOf('CardPlaylist', module);

const data = [
  {
    title: 'Bailando',
    image: 'https://i.scdn.co/image/723b3060add0aff0011fd2ae7d59237bbe788bb2',
    songs: 60
  },
  {
    title: 'Esquenta Sertanejo',
    image: 'https://i.scdn.co/image/ac3e0fa062dbd40285b288fb7b17625dcd1948e1',
    songs: 50
  },
  {
    title: 'Funk 25',
    image: 'https://i.scdn.co/image/f45205a65ddcca6ce2eef1cbc93c84fbc2503e72',
    songs: 25
  }
];

stories.add('without props', () => <CardPlaylist data={data} />);
