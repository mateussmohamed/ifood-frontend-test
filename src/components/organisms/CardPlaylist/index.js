import React from 'react';
import PropTypes from 'prop-types';
import Card from '../../molecules/Card';

import './CardPlaylist.css';

const CardPlaylist = ({ data }) => (
  <section className="card-playlist">
    {data.map(item => (
      <Card
        key={item.id}
        title={item.name}
        image={item.images[0].url}
        songs={item.tracks.total}
      />
    ))}
  </section>
);

CardPlaylist.defaultProps = {
  data: []
};

CardPlaylist.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object)
};

export default CardPlaylist;
