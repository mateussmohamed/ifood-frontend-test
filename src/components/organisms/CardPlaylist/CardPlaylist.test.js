import React from 'react';
import renderer from 'react-test-renderer';
import CardPlaylist from './index';

it('Should CardPlaylist match to snapshot', () => {
  const component = renderer.create(
    <CardPlaylist data={[]}>TÃ­tulo</CardPlaylist>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
