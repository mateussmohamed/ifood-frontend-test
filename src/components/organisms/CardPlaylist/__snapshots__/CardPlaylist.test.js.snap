// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CardPlaylist match to snapshot 1`] = `
<section
  className="card-playlist"
/>
`;
